src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,179 @@
+            updatePreparedStatement.setString(2, car.getModel());
+            updatePreparedStatement.setLong(3, car.getId());
+            updatePreparedStatement.executeUpdate();
+            deletePreparedStatement.setLong(1, car.getId());
+            deletePreparedStatement.executeUpdate();
+            insertPreparedStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertPreparedStatement.setLong(2, driver.getId());
+                insertPreparedStatement.executeUpdate();
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(deleteQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot deleted car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverQuery = "SELECT * FROM cars c"
+                + " INNER JOIN cars_drivers cd ON cd.car_id = c.id"
+                + " INNER JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE cd.driver_id = ? AND m.deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection
+                         .prepareStatement(getAllByDriverQuery)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get any car by driver by id " + driverId, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("c.id", Long.class);
+        String model = resultSet.getObject("c.model", String.class);
+        Long manufacturerId = resultSet.getObject("c.manufacturer_id", Long.class);
+        String nameManufacturer = resultSet.getObject("m.name", String.class);
+        String countryManufacturer = resultSet.getObject("m.country", String.class);
+        Manufacturer manufacturer = new Manufacturer(nameManufacturer, countryManufacturer);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car(model, manufacturer);
+        car.setId(id);
+        car.setDrivers(getDrivers(id));
```

Comment:
here you are opening one connection inside of another - this is not good
let's return the car with manufacturer only here